run_public_tests.py

```
@@ -0,0 +1,195 @@
+
+    def run_test(self, test_path, test_name):
+        """Run a single test file"""
+        print("\n" + "=" * 70)
+        print(f"Running: {test_name}")
+        print("=" * 70)
+
+        try:
+            env = os.environ.copy()
+            env["PYTHONPATH"] = str(self.testing_dir)
+
+            result = subprocess.run(
```

Ruff (0.14.7)

`subprocess` call: check for execution of untrusted input

(S603)